prov/verbs/src/verbs_info.c

```diff
@@ -1760,7 +1771,7 @@ int vrb_getinfo(uint32_t version, const char *node, const char *service,
 
 	ofi_alter_info(*info, hints, version);
 
-	vrb_alter_info(hints, *info);
+	vrb_alter_info(hints, *info, version);
```

Comment:
It still seems that something isn't right here.  ofi_alter_info() *should* have handled this, so the lower-level providers don't need to.

I think the issue might be in fi_alter_domain_attr() (util_attr.c).  Original code is below

```
	if (hints_mr_mode & (FI_MR_BASIC | FI_MR_SCALABLE)) {
		attr->mr_mode = hints_mr_mode;
	} else if (FI_VERSION_LT(api_version, FI_VERSION(1, 5))) {
		attr->mr_mode = (attr->mr_mode && attr->mr_mode != FI_MR_SCALABLE) ?
				FI_MR_BASIC : FI_MR_SCALABLE;
	} else {     
		if ((hints_mr_mode & attr->mr_mode) != attr->mr_mode) {
			attr->mr_mode = ofi_cap_mr_mode(info_caps,
						attr->mr_mode & hints_mr_mode);
		}
	}
```
I think we want to change the else part to:
```
} else {
    attr->mr_mode &= ~(FI_MR_BASIC | FI_MR_SCALABLE);
    if ((hints_mr_mode & attr->mr_mode) != attr->mr_mode) {
```
to ensure that those bits are cleared for version > 1.5, unless the app had requested them.

Reply:
Looking at ofi_alter_info(), the reason why it is not being run for this case is because I am providing NULL hints. ofi_alter_info() won't call fi_alter_domain_attr() to adjust MR mode bits. This behavior seems incorrect.

Reply:
I agree.  The fi_alter_[domain, ep, rx, tx]_attr() functions that ofi_alter_info() calls handle null hints and adjust the attributes even in those cases.  Can you update ofi_alter_info() to handle null hints within the for loop, with the suggested update above?

Reply:
Will do.